server/index-node/src/explorer.rs

```diff
@@ -121,14 +153,32 @@ where
         Ok(as_http_response(&value))
     }
 
-    fn handle_entity_count(&self, deployment: &str) -> Result<Response<Body>, GraphQLServerError> {
+    fn handle_entity_count(
+        &self,
+        logger: &Logger,
+        deployment: &str,
+    ) -> Result<Response<Body>, GraphQLServerError> {
+        let start = Instant::now();
         if let Some(value) = self.entity_counts.get(deployment) {
+            if start.elapsed() > *LOCK_THRESHOLD {
```

Comment:
Lets move the check outside of the if statement so that we check the elapsed duration even on misses.